compile error on "ucs_memory_type_t"

Please submit all the information below so that we can understand the working environment that is the context for your question.

## Background information

### What version of Open MPI are you using? (e.g., v4.1.6, v5.0.1, git branch name and hash, etc.)

v4.1.7

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

from a source

### If you are building/installing from a git clone, please copy-n-paste the output from `git submodule status`.

download from official website

### Please describe the system on which you are running

Linux version 3.10.0-1160.el7.x86_64
Intel(R) Xeon(R) Gold 5122 CPU

-----------------------------

## Details of the problem

when compile with gcc 12, 

../configure CC=gcc CXX=g++  --prefix=...

I get the follow errors:

../../../../../oshmem/mca/sshmem/ucx/sshmem_ucx_module.c:99:41: error: unknown type name 'ucs_memory_type_t'
   99 |                         unsigned flags, ucs_memory_type_t mem_type, int err_level)
      |                                         ^~~~~~~~~~~~~~~~~
../../../../../oshmem/mca/sshmem/ucx/sshmem_ucx_module.c: In function 'segment_create':
../../../../../oshmem/mca/sshmem/ucx/sshmem_ucx_module.c:205:12: warning: implicit declaration of function 'segment_create_internal' [-Wimplicit-function-declaration]
  205 |     return segment_create_internal(ds_buf, mca_sshmem_base_start_address, size,
      |            ^~~~~~~~~~~~~~~~~~~~~~~
../../../../../oshmem/mca/sshmem/ucx/sshmem_ucx_module.c:206:43: error: 'UCS_MEMORY_TYPE_HOST' undeclared (first use in this function)
  206 |                                    flags, UCS_MEMORY_TYPE_HOST, 0);

Thanks in advance ~

--- a/config/ompi_check_ucx.m4
+++ b/config/ompi_check_ucx.m4
@@ -61,28 +61,18 @@ AC_DEFUN([OMPI_CHECK_UCX],[
 
     AS_IF([test "$ompi_check_ucx_happy" = yes],
           [# Turn off UCX version v1.8 due to issue #8321
-           AC_CACHE_CHECK([UCX version 1.8.x],
-               [ompi_check_ucx_cv_have_version_1_8],
+           AC_CACHE_CHECK([UCX version > 1.9.x],
+               [ompi_check_ucx_cv_have_version_gt_1_9],
                [AC_PREPROC_IFELSE([AC_LANG_PROGRAM([[
 #include <ucp/api/ucp_version.h>
                                  ]], [[
-#if (UCP_API_MAJOR == 1) && (UCP_API_MINOR == 8)
-#error "Invalid version"
-#endif
-                                 ]])],
-                             [ompi_check_ucx_cv_have_version_1_8=no],
-                             [ompi_check_ucx_cv_have_version_1_8=yes])])
-           AS_IF([test "${ompi_check_ucx_cv_have_version_1_8}" = "yes"],
-                 [AC_MSG_WARN([UCX support skipped because version 1.8.x was found, which has a known catastrophic issue.])
-                  ompi_check_ucx_happy=no])])
-           AC_PREPROC_IFELSE([AC_LANG_PROGRAM([[
-#include <ucp/api/ucp_version.h>
-                             ]], [[
 #if (UCP_API_MAJOR < 1) || ((UCP_API_MAJOR == 1) && (UCP_API_MINOR < 9))
 #error "Version too low"
 #endif
-                             ]])],
-                             [], [AC_MSG_WARN([UCX version is too old, please upgrade to 1.9 or higher.])])
+                                 ]])],
+                             [ompi_check_ucx_cv_have_version_gt_1_9=yes],
+                             [ompi_check_ucx_cv_have_version_gt_1_9=no
+                              ompi_check_ucx_happy=no])])])
 
     AS_IF([test "$ompi_check_ucx_happy" = yes],
           [AC_CHECK_DECLS([ucp_tag_send_nbr],

